src/pymor/reductors/loewner.py

```diff
@@ -0,0 +1,331 @@
+        The ordering with respect to which the partitioning rule is executed. Can be either
+        'magnitude', 'random' or 'regular'. Defaults to 'regular'.
+    conjugate
+        Whether to guarantee realness of reduced |LTIModel| by keeping complex conjugates in the
+        same partitioning or not. If `True` will automatically generate conjugate data if necessary.
+    mimo_handling
+        Option indicating how to treat MIMO systems. Can be:
+
+        - `'random'` for using random tangential directions.
+        - `'full'` for fully interpolating all input-output pairs.
+        - Tuple `(ltd, rtd)` where `ltd` corresponds to left and `rtd` to right tangential
+          directions.
+    """
+
+    cache_region = 'memory'
+
+    def __init__(self, s, Hs, partitioning='even-odd', ordering='regular', conjugate=True, mimo_handling='random'):
+        assert isinstance(s, np.ndarray)
+        if hasattr(Hs, 'transfer_function'):
+            Hs = Hs.transfer_function
+        assert isinstance(Hs, (TransferFunction, np.ndarray, list))
+
+        assert partitioning in ('even-odd', 'half-half') \
+            or len(partitioning) == 2 \
+            and len(partitioning[0]) + len(partitioning[1]) == len(s)
+        assert ordering in ('magnitude', 'random', 'regular')
+
+        if isinstance(Hs, TransferFunction):
+            Hss = np.empty((len(s), Hs.dim_output, Hs.dim_input), dtype=s[0].dtype)
+            for i, ss in enumerate(s):
+                Hss[i] = Hs.eval_tf(ss)
+            Hs = Hss
+        else:
+            Hs = Hs
+            assert Hs.shape[0] == len(s)
+
+        # ensure that complex sampling values appear in complex conjugate pairs
+        if conjugate:
+            # if user provides partitioning sizes, make sure they are adjusted
+            if isinstance(partitioning, tuple):
+                p0 = partitioning[0]
+                p1 = partitioning[1]
+                for i, ss in enumerate(s):
+                    if np.conj(ss) not in s:
+                        s = np.append(s, np.conj(ss))
+                        Hs = np.append(Hs, np.conj(Hs[i])[np.newaxis, ...], axis=0)
+                        if i in p0:
+                            p0 = np.append(p0, len(s)-1)
+                        else:
+                            p1 = np.append(p1, len(s)-1)
+                partitioning = (p0, p1)
+            else:
+                for i, ss in enumerate(s):
+                    if np.conj(ss) not in s:
+                        s = np.append(s, np.conj(ss))
+                        Hs = np.append(Hs, np.conj(Hs[i])[np.newaxis, ...], axis=0)
+
+        if len(Hs.shape) > 1:
+            self.dim_output = Hs.shape[1]
+            self.dim_input = Hs.shape[2]
+            if self.dim_output == self.dim_output == 1:
+                Hs = np.squeeze(Hs)
+        else:
+            self.dim_output = 1
+            self.dim_input = 1
+
+        self.__auto_init(locals())
+
+    def reduce(self, r=None, tol=1e-7):
```

Comment:
I would be for `tol=0` to be the default, since I feel like the priority should be on interpolation, and the user should be the one to decide about the truncation tolerance. Is there a specific reason to use `1e-7`?

Reply:
Yes, I agree it's a good idea (maybe even necessary) to truncate when there is too much data. But, Loewner can also be used in situations with too little data, e.g., in TF-IRKA and in methods for computing the $\mathcal{H}_\infty$ norm, and then the goal is to have interpolation.

I'm not sure what is usually meant by "Loewner"; if it's only interpolation, or also truncation. I suppose it's not a bad idea to check for ill-conditioning, so using a small positive number for the `tol` by default could be ok. Also, in `gram_schimidt` and `pod` there are already default positive tolerances.